packages/graph/hash_graph/lib/graph/src/store/query/mod.rs

```diff
@@ -68,10 +85,16 @@ fn compare(lhs: &Literal, rhs: &Literal) -> Result<bool, ExpressionError> {
         (lhs, Literal::List(rhs)) => rhs.iter().try_find(|rhs| compare(lhs, rhs))?.is_some(),
 
         // Version
-        (Literal::Version(lhs, _), Literal::Float(rhs)) => *lhs == *rhs as u32,
-        (Literal::Float(lhs), Literal::Version(rhs, _)) => *lhs as u32 == *rhs,
-        (Literal::String(lhs), Literal::Version(_, latest)) if lhs == "latest" => *latest,
+        (Literal::Version(Version::Ontology(lhs), _), Literal::Float(rhs)) => *lhs == *rhs as u32,
+        (Literal::Version(Version::Entity(lhs), _), Literal::Float(rhs)) => {
+            lhs.timestamp() == *rhs as i64
+        }
         (Literal::Version(_, latest), Literal::String(rhs)) if rhs == "latest" => *latest,
+        (Literal::Float(lhs), Literal::Version(Version::Ontology(rhs), _)) => *lhs as u32 == *rhs,
+        (Literal::Float(lhs), Literal::Version(Version::Entity(rhs), _)) => {
```

Comment:
What do you think about combining the commuted variants together as such:
```rust
  (Literal::Float(float), Literal::Version(Version::Entity(entity_version_id), _)) 
  | (Literal::Version(Version::Entity(entity_version_id), _), Literal::Float(float)) => {
    entity_version_id.timestamp() == *float as i64
```
so we don't have slightly different match arms for the same case?

Reply:
Good point, applied in 673a6eb0fdea7c6559744b8e30b75a4732d59169